packages/jaeger-ui/src/components/TracePage/TracePageHeader/TracePageHeader.tsx

```diff
@@ -91,7 +91,23 @@ export const HEADER_ITEMS = [
   {
     key: 'span-count',
     label: 'Total Spans',
-    renderer: (trace: Trace) => trace.spans.length,
+    renderer: (trace: Trace) => {
+      if (trace.hasOrphanSpans) {
+        const tooltipText = `This trace may be incomplete. ${trace.orphanSpanCount} span${trace.orphanSpanCount !== 1 ? 's' : ''} ${trace.orphanSpanCount !== 1 ? 'have' : 'has'} missing parent span(s).`;
+        return (
+          <span>
+            {trace.spans.length}{' '}
+            <Tooltip title={tooltipText}>
+              <span className="TracePageHeader--incompleteTag">
+                <IoWarning className="TracePageHeader--incompleteIcon" />
+                Incomplete
+              </span>
+            </Tooltip>
+          </span>
+        );
+      }
+      return trace.spans.length;
```

Comment:
this is confusing organization, the DOM snippet should be returned only once and the conditional be inside it, e.g.
```
return (
  <span>
    {trace.spans.length}
    (trace.orphanSpans > 0 && (<span>...))
)
```